assets/scripts/dark-mode.js

```diff
@@ -0,0 +1,66 @@
+class DarkMode extends HTMLElement {
+  constructor() {
+    super();
+    this.attachShadow({ mode: "open" });
+  }
+
+  connectedCallback() {
+    this.render();
+    this.setDarkMode();
+  }
+
+  setDarkMode() {
```

Comment:
I think there's a slightly simpler way to do this too. How about something more like

```js
toggleDarkMode = () => {
    this.isDarkMode = !this.isDarkMode;
    localStorage.setItem("isDarkMode", this.isDarkMode ? "1" : "0");
    this.updateBodyClass();
  };

  updateBodyClass() {
    document.body.classList.toggle("is-dark-mode", this.isDarkMode);
    document.body.classList.toggle("is-light-mode", !this.isDarkMode);
  }
```